safenode/src/network/mod.rs

```diff
@@ -0,0 +1,152 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+mod api;
+mod command;
+mod error;
+mod event;
+mod safe_msg;
+
+pub use self::{
+    event::NetworkEvent,
+    safe_msg::{SafeRequest, SafeResponse},
+};
+
+use self::{
+    api::NetworkApi,
+    command::CmdToSwarm,
+    error::Result,
+    event::SafeNodeBehaviour,
+    safe_msg::{SafeMsgCodec, SafeMsgProtocol},
+};
+use futures::{
+    channel::{mpsc, oneshot},
+    prelude::*,
+};
+use libp2p::{
+    core::muxing::StreamMuxerBox,
+    identity,
+    kad::{record::store::MemoryStore, Kademlia, KademliaConfig, QueryId},
+    mdns,
+    request_response::{self, ProtocolSupport, RequestId},
+    swarm::{Swarm, SwarmBuilder},
+    PeerId, Transport,
+};
+use std::{
+    collections::{HashMap, HashSet},
+    iter,
+    time::Duration,
+};
+use tracing::warn;
+
+/// The main event loop recieves `SwarmEvents` from the network, `CmdToSwarm` from the upper layers and
+/// emmits `NetworkEvent` to the upper layers
+/// Also keeps track of the pending queries/requests and their channels. Once we recieve an event
+/// that is the outcome of a previously executed cmd, send a response to them via the stored channel.
+pub struct EventLoop {
+    swarm: Swarm<SafeNodeBehaviour>,
+    command_receiver: mpsc::Receiver<CmdToSwarm>,
+    event_sender: mpsc::Sender<NetworkEvent>,
+    pending_dial: HashMap<PeerId, oneshot::Sender<Result<()>>>,
+    pending_start_providing: HashMap<QueryId, oneshot::Sender<Result<()>>>,
+    pending_get_providers: HashMap<QueryId, oneshot::Sender<HashSet<PeerId>>>,
+    pending_safe_requests: HashMap<RequestId, oneshot::Sender<Result<SafeResponse>>>,
+}
+
+impl EventLoop {
```

Comment:
I think `EventLoop` doesn't really capture the fact that we start up the connections in here.

Perhaps `NetworkSwarmLoop` ?